extensions/sql-migration/src/dashboard/tabBase.ts

```diff
@@ -136,6 +142,43 @@ export abstract class TabBase<T> implements azdata.Tab, vscode.Disposable {
 		return newMigrationButton;
 	}
 
+	protected createImportMigrationButton(): azdata.ButtonComponent {
+		const importMigrationButton = this.view.modelBuilder.button()
+			.withProps({
+				buttonType: azdata.ButtonType.Normal,
+				label: loc.DESKTOP_IMPORT_MIGRATION_BUTTON_LABEL,
+				description: loc.DESKTOP_IMPORT_MIGRATION_BUTTON_DESCRIPTION,
+				height: 24,
+				iconHeight: 24,
+				iconWidth: 24,
+				iconPath: IconPathHelper.addNew,
+			}).component();
+		this.disposables.push(
+			importMigrationButton.onDidClick(async () => {
+				const filepath = await utils.promptUserForFile({ 'Json (*.json)': ['json'] });
+				if (filepath) {
+					try {
+						const assessmentReportJson = fs.readFileSync(filepath, 'utf-8');
+						const assessmentReport = JSON.parse(assessmentReportJson);
+
+						const serverName = (await getSourceConnectionProfile()).serverName;
+						const saveInfo = parseAssessmentReport(assessmentReport);
+						await this.context.globalState.update(`${this.mementoToken}.${serverName}`, saveInfo);
+					} catch (err) {
+						void vscode.window.showInformationMessage(`Selected invalid format import file: ${filepath}`);
```

Comment:
it might be other errors? for example, the json file is old assessment report generated from DMA. The assessment models are different.

Reply:
Import feature support the DMA reports as well.

Reply:
I mean why not using err.message to show customers the root cause?

Reply:
makes sense. Fixed.